extensions/vault/runtime/src/main/java/io/quarkus/vault/config/CredentialProviderConfig.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.vault.config;
+
+import static io.quarkus.vault.CredentialProvider.PASSWORD_PROPERTY_NAME;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class CredentialProviderConfig {
+
+    public CredentialProviderConfig() {
+    }
+
+    public CredentialProviderConfig(String dbrole, String path, String key) {
+        this.dbrole = dbrole;
+        this.path = path;
+        this.key = key;
+    }
+
+    /**
+     * dbrole
```

Comment:
We need better documentation. We can now autogenerate the documentation from the Javadoc comments so better have them good.

BTW, why `dbrole`. Is it a Vault term?

Reply:
I improved it. It was one of the last part we worked on with sergey. and that was missing some attention.
I provided as well better names for properties. 
dbrole is now database-creds-role, which is the terminology used in https://www.vaultproject.io/docs/secrets/databases/index.html
path and key and now kvPath and KvKey to make sure it is clear that the kv* stuff is separate from the dynamic db credentials.
this is also reinforced in documentation.